src/libraries/System.Net.Connections/src/System/Net/Connections/Sockets/SocketConnection.cs

```diff
@@ -13,34 +13,38 @@ namespace System.Net.Connections
 {
     internal sealed class SocketConnection : Connection, IConnectionProperties
     {
-        private readonly NetworkStream _stream;
+        private readonly Socket _socket;
+        private readonly SocketsConnectionFactory _factory;
+        private readonly IConnectionProperties? _options;
+        private Stream? _stream;
+        private IDuplexPipe? _pipe;
 
-        public override EndPoint? RemoteEndPoint => _stream.Socket.RemoteEndPoint;
-        public override EndPoint? LocalEndPoint => _stream.Socket.LocalEndPoint;
+        public override EndPoint? RemoteEndPoint => _socket.RemoteEndPoint;
+        public override EndPoint? LocalEndPoint => _socket.LocalEndPoint;
         public override IConnectionProperties ConnectionProperties => this;
 
-        public SocketConnection(Socket socket)
+        public SocketConnection(Socket socket, SocketsConnectionFactory factory, IConnectionProperties? options)
         {
-            _stream = new NetworkStream(socket, ownsSocket: true);
+            _socket = socket;
+            _factory = factory;
+            _options = options;
         }
 
         protected override ValueTask CloseAsyncCore(ConnectionCloseMethod method, CancellationToken cancellationToken)
         {
-            if (cancellationToken.IsCancellationRequested)
-            {
-                return ValueTask.FromCanceled(cancellationToken);
-            }
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This should be reverted; it makes the exception come from the method rather than be bundled inside the `ValueTask`, which won't be expected by users.